tests/storage/migration.go

```diff
@@ -747,7 +747,7 @@ var _ = Describe(SIG("Volumes update with migration", decorators.RequiresTwoSche
 					return controller.NewVirtualMachineConditionManager().HasCondition(
 						vm, virtv1.VirtualMachineManualRecoveryRequired)
 				}).WithTimeout(120 * time.Second).WithPolling(time.Second).Should(BeFalse())
-				libwait.WaitForVMIPhase(vmi, []v1.VirtualMachineInstancePhase{v1.Running})
+				Eventually(matcher.ThisVMI(vmi), 360*time.Second, time.Second).Should(matcher.BeInPhase(v1.Running))
```

Comment:
I fail to understand how this is related.
Should these failures concern us that some behavior was changed in this PR?

Reply:
@iholder101 no, the WaitForVMIPhase looks at weird warnings and fails on them. For example `Unexpected Warning event received: testvmi-klbkg,f35c019b-b555-4665-a08f-ea219efa6a63: Failed to delete virtual machine pod virt-launcher-testvmi-klbkg-qn8zd
Obviously, all pods are deleted in this test, and it is complete. I think the warning is related to garbage collection. I was watching the pods manually, there were no delays or failures.

Reply:
This is most certainly not related to your changes, but still concerning.
I've opened an issue to track this: https://github.com/kubevirt/kubevirt/issues/14326.